src/client/network/mod.rs

```diff
@@ -394,36 +440,76 @@ enum NodeHealth {
     ///
     /// Once we've reached `healthyAt` the node is *semantically* in the ``unused`` state,
     /// other than retaining the backoff until a `healthy` request happens.
-    Unhealthy { backoff_interval: Duration, healthy_at: Instant },
+    Unhealthy { backoff: NodeBackoff, healthy_at: Instant, attempts: usize },
 
     /// When we last used the node the node acted as normal, so, we get to treat it as a healthy node for 15 minutes.
     Healthy { used_at: Instant },
 }
 
+#[derive(Copy, Clone)]
+pub(crate) struct NodeBackoff {
+    pub(crate) current_interval: Duration,
+    pub(crate) max_backoff: Duration,
+    pub(crate) min_backoff: Duration,
+    pub(crate) max_attempts: usize,
```

Comment:
Can we make this an option?

Reply:
These fields have default values upon initialization. Also, the function that handles exponential backoff doesn't have optional fields for the common fields listed in the `NodeBackoff` struct.

Reply:
Right, but "0 = unlimited", it might make more sense to have this as an `Option<NonZeroUsize>`, where `None` = "unlimited" and anything else is the limit, the function that handles the backoff can say `if max_attempts.map_or(false, |max_attempts| attempts > max_attempts.get())` for the removal.